mne/io/meas_info.py

```diff
@@ -1887,15 +1893,90 @@ def anonymize_info(info):
     Operates in place.
     """
     _validate_type(info, 'info', "self")
-    if info.get('subject_info') is not None:
-        del info['subject_info']
-    info['meas_date'] = None
+
+    default_anon_dos = datetime.datetime(2000, 1, 1, 0, 0, 0)
+    default_str = "mne_anonymize"
+    default_subject_id = 0
+    default_desc = ("Anonymized using a time shift"
+                    " to preserve age at acquisition")
+
+    if timeshift is None:
+        tmp = (info['meas_date'][0] -
+               datetime.datetime.timestamp(default_anon_dos))
+        delta_t = datetime.timedelta(seconds=tmp)
+    else:
+        delta_t = datetime.timedelta(days=timeshift)
```

Comment:
I would create a private function call `_anonymize_date` that takes dateback param and is called on each date and returns the fixed date. You can then call the same function on annotations.orig_time